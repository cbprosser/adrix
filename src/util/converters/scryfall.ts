import { APICard } from '../../types';

export const cardToMUIRow = ({
  rarity,
  set,
  prices: { usd, tix, eur, ...restPrices },
  ...other
}: APICard) => ({
  eur: eur ? +eur : undefined,
  rarity: `${rarity[0].toUpperCase()}${rarity.slice(1)}`,
  set: set.toUpperCase(),
  tix: tix ? +tix : undefined,
  usd: usd ? +usd : undefined,
  prices: { usd, tix, eur, ...restPrices },
  ...other,
});

export const muiRowToCard = ({
  eur,
  rarity,
  set,
  tix,
  usd,
  ...other
}: ReturnType<typeof cardToMUIRow>) =>
  ({
    rarity: `${rarity[0].toLowerCase()}${rarity.slice(1)}`,
    set: set.toLowerCase(),
    ...other,
  } as APICard);
