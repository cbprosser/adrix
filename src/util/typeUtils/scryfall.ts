export const relatedComponents = [
  'combo_piece',
  'meld_part',
  'meld_result',
  'token',
] as const;

export const rarities = [
  'bonus',
  'common',
  'mythic',
  'rare',
  'special',
  'uncommon',
] as const;

export const imageVersions = [
  'small',
  'normal',
  'large',
  'png',
  'art_crop',
  'border_crop',
] as const;

export const multisideLayouts = [
  'double_faced_token',
  'double_sided',
  'modal_dfc',
  'transform',
] as const;

export const multifaceLayouts = [
  'adventure',
  'flip',
  'split',
  ...multisideLayouts,
] as const;

export const layouts = [
  'art_series',
  'augment',
  'emblem',
  'host',
  'leveler',
  'meld',
  'normal',
  'planar',
  'saga',
  'scheme',
  'token',
  'vanguard',
  ...multifaceLayouts,
] as const;

export const colorAbbrvs = ['W', 'U', 'B', 'R', 'G'] as const;

export const legalities = [
  'banned',
  'legal',
  'not_legal',
  'restricted',
] as const;

export const finishes = ['etched', 'glossy', 'nonfoil', 'foil'] as const;

export const securityStamps = ['oval', 'triangle', 'acorn', 'arena'] as const;
