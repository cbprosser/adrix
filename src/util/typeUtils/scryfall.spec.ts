import * as scryfallTypesObjects from './scryfall';

describe('scryfall types objects Suite', () => {
  it("Should match what's expected", () => {
    expect(scryfallTypesObjects).toMatchSnapshot();
  });
});
