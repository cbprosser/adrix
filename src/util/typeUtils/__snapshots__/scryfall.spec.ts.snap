// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scryfall types objects Suite Should match what's expected 1`] = `
Object {
  "colorAbbrvs": Array [
    "W",
    "U",
    "B",
    "R",
    "G",
  ],
  "finishes": Array [
    "etched",
    "glossy",
    "nonfoil",
    "foil",
  ],
  "imageVersions": Array [
    "small",
    "normal",
    "large",
    "png",
    "art_crop",
    "border_crop",
  ],
  "layouts": Array [
    "art_series",
    "augment",
    "emblem",
    "host",
    "leveler",
    "meld",
    "normal",
    "planar",
    "saga",
    "scheme",
    "token",
    "vanguard",
    "adventure",
    "flip",
    "split",
    "double_faced_token",
    "double_sided",
    "modal_dfc",
    "transform",
  ],
  "legalities": Array [
    "banned",
    "legal",
    "not_legal",
    "restricted",
  ],
  "multifaceLayouts": Array [
    "adventure",
    "flip",
    "split",
    "double_faced_token",
    "double_sided",
    "modal_dfc",
    "transform",
  ],
  "multisideLayouts": Array [
    "double_faced_token",
    "double_sided",
    "modal_dfc",
    "transform",
  ],
  "rarities": Array [
    "bonus",
    "common",
    "mythic",
    "rare",
    "special",
    "uncommon",
  ],
  "relatedComponents": Array [
    "combo_piece",
    "meld_part",
    "meld_result",
    "token",
  ],
}
`;
